From 42b180600937c0396ca81a3606e1c87a455e39ca Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Mon, 5 Feb 2018 14:57:28 +0100
Subject: [PATCH] build: Disable seccomp on linux architectures where it's not
 supported

seccomp is currently not supported on all the linux architectures, see:
https://github.com/seccomp/libseccomp/blob/master/src/arch.c

Only disable seccomp on architectures where it's not supported, keep it
enabled on all the other (linux) ones.

https://bugzilla.gnome.org/show_bug.cgi?id=786358
---
 configure.ac | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 136f97fb..42e99542 100644
--- a/configure.ac
+++ b/configure.ac
@@ -160,14 +160,24 @@ else
 fi
 
 SECCOMP_PKG=""
+enable_seccomp="no"
 dnl Check for bubblewrap compatible platform
 case $host_os in
   linux*)
-    PKG_CHECK_MODULES(LIBSECCOMP, [libseccomp])
-    SECCOMP_PKG="libseccomp"
+    case $host_cpu in
+      *)
+        PKG_CHECK_MODULES(LIBSECCOMP, [libseccomp])
+        SECCOMP_PKG="libseccomp"
+        AC_DEFINE([ENABLE_SECCOMP], [1], [Define if using seccomp])
+        enable_seccomp="yes"
+        ;;
+      alpha|ia64|m68k|sh4|sparc64)
+        enable_seccomp="no (not avaiable on this architecture)"
+        AC_MSG_WARN("seccomp not available on this architecture")
+        ;;
+    esac
     AC_DEFINE_UNQUOTED(_GNU_SOURCE, 1, [Define to include GNU extensions])
     AC_DEFINE_UNQUOTED(HAVE_BWRAP, 1, [Define to 1 if Bubblewrap support is available])
-    AC_DEFINE([ENABLE_SECCOMP], [1], [Define if using seccomp])
     AC_DEFINE_UNQUOTED(INSTALL_PREFIX, "$prefix", [Path to library install prefix])
     ;;
 esac
@@ -270,5 +280,6 @@ echo "
         Date in gnome-version.xml:    ${enable_date_in_gnome_version}
         Build gtk-doc documentation:  ${enable_gtk_doc}
         Build with udev support:      ${enable_udev}
+        Build with seccomp support:   ${enable_seccomp}
 
 "
-- 
2.15.1

